File: task1/src/main/java/framework/baseentity/BasePage.java
package framework.baseentity;

import framework.elements.Panel;
import org.openqa.selenium.By;

public abstract class BasePage {

    private String namePage;
    private By locator;

    public BasePage(String namePage, By locator) {
        this.namePage = namePage;
        this.locator = locator;
    }

    public boolean isPresentPage() {
        return new Panel(locator, namePage).isDisplayedElement();
    }

    public String getNamePage() {
        return namePage;
    }

    public String getFullElementName(String nameElement) {
        return nameElement + " on " + namePage;
    }
}

File: ButtleShip/src/main/kotlin/framework/pages/BasePage.kt
package framework.pages

import framework.elements.Label
import framework.utils.Logger
import org.openqa.selenium.By

abstract class BasePage(val locator: By, val namePage: String) {

    private var logger: Logger = Logger()

    init {
        logger.info("Create page $namePage")
    }

    fun isPresent(): Boolean {
        logger.info("Is displayed page $namePage")
        return getPageLabel().isDisplayed()
    }

    private fun getPageLabel(): Label {
        return Label(locator, namePage)
    }
}

